File: student-frontend/src/Services/FacultyService.js
import axios from "axios";
const BACKEND_LINK="http://localhost:8080/faculty"

class FacultyService
{
    addFaculty(faculty)
    {
        return axios.post(BACKEND_LINK+"/addFaculty",faculty);
    }
    findFaculty(id)
    {
        return axios.get(BACKEND_LINK+"/findFaculty/"+id);
    }
    enrollStudent(facultyId,studentId)
    {
        return axios.put(BACKEND_LINK+"/enroll/"+facultyId+"/"+studentId);
    }
    changePassword(facultyId,oldP,newP)
    {
        return axios.put(BACKEND_LINK+"/changePassword/"+facultyId+"/"+oldP+"/"+newP);
    }

}
export default new FacultyService();

File: student-backend/src/main/java/com/minor1/StudentManagement/service/FacultyService.java
package com.minor1.StudentManagement.service;

import com.minor1.StudentManagement.entities.Faculty;
import com.minor1.StudentManagement.exception.OperationFailedException;
import com.minor1.StudentManagement.exception.ResourceNotFoundException;

public interface FacultyService {
	
	Faculty addFaculty(Faculty faculty) throws OperationFailedException;
	
	Faculty findFaculty(long facultyId) throws OperationFailedException, ResourceNotFoundException;
	
	void enrollStudent(long facultyId, long studentId) throws OperationFailedException, ResourceNotFoundException;
	
	String changePassword(long facultyId, String oldPassword, String newPassword) throws ResourceNotFoundException;

}
